Fika.Core/Coop/Utils/FikaBackendUtils.cs

```diff
@@ -69,18 +71,31 @@ public static bool JoinMatch(string profileId, string serverId, out CreateMatch
                 return false;
             }
 
+            FikaVersionLabelUpdate_Patch.raidCode = result.RaidCode;
+            
             return true;
         }
 
         public static void CreateMatch(string profileId, string hostUsername, RaidSettings raidSettings)
         {
             long timestamp = DateTimeOffset.Now.ToUnixTimeSeconds();
-            var body = new CreateMatch(profileId, hostUsername, timestamp, raidSettings, HostExpectedNumberOfPlayers, raidSettings.Side, raidSettings.SelectedDateTime);
+            var raidCode = GenerateRaidCode(6);
+            var body = new CreateMatch(raidCode, profileId, hostUsername, timestamp, raidSettings, HostExpectedNumberOfPlayers, raidSettings.Side, raidSettings.SelectedDateTime);
 
             FikaRequestHandler.RaidCreate(body);
 
             SetGroupId(profileId);
             MatchingType = EMatchmakerType.GroupLeader;
+            
+            FikaVersionLabelUpdate_Patch.raidCode = raidCode;
+        }
+
+        private static string GenerateRaidCode(int length)
+        {
+            var random = new Random();
+            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+            return new string(Enumerable.Repeat(chars, length)
```

Comment:
Is there another method that doesn't use `LINQ`? Otherwise leave it, it's fine as it is.

Reply:
solved, thks.